code/modules/library/book.dm

```diff
@@ -167,7 +167,7 @@
 		user << browse("<body bgcolor='[book_bgcolor]'>[dat]<br>" + "[pages[current_page]]", "window=book[UID()]")
 
 /obj/item/book/Topic(href, href_list)
-	if(..())
+	if(..() || isobserver(usr))
 		return
```

Comment:
Yeah dont do this.

I would rather ghosts be unable to turn pages as opposed to ghosts being able to turn pages while someone is reading the book.